packages/cdk-graph-plugin-threat-composer/src/plugin.ts

```diff
@@ -0,0 +1,124 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import * as fs from "fs";
+import {
+  CdkGraph,
+  CdkGraphArtifact,
+  CdkGraphContext,
+  ICdkGraphPlugin,
+  IGraphPluginBindCallback,
+  IGraphReportCallback,
+} from "@aws/cdk-graph";
+import { PDKNagApp } from "@aws/pdk-nag";
+import { ThreatModelGenerator } from "./model-generator/threat-model-generator";
+import { ThreatComposerApplicationDetails } from "./model-generator/types";
+
+/**
+ * Options for the Threat Composer CDK Graph plugin
+ */
+export interface CdkGraphThreatComposerPluginOptions {
+  /**
+   * Details about the application to include in the threat model
+   */
+  readonly applicationDetails?: ThreatComposerApplicationDetails;
+}
+
+/**
+ * CdkGraphThreatComposerPlugin is a {@link ICdkGraphPlugin CdkGraph Plugin} implementation for generating
+ * Threat Composer threat models.
+ * @see https://github.com/awslabs/threat-composer
+ */
+export class CdkGraphThreatComposerPlugin implements ICdkGraphPlugin {
+  /**
+   * Fixed ID of the threat-composer plugin
+   */
+  static readonly ID = "threat-composer";
+
+  /**
+   * Curent semantic version of the threat-composer plugin
+   */
+  static readonly VERSION = "0.0.0";
```

Comment:
How will this be maintained?

Reply:
I think this is only for cdk graph plugin dependencies - it's a required property so I followed suit with the diagram plugin. From a quick look at the cdk-graph code I couldn't see how dependencies were honoured within the plugin framework though, but I might not have looked hard enough!